FEXCore/Source/Interface/Core/OpcodeDispatcher.h

```diff
@@ -869,8 +869,9 @@ friend class FEXCore::IR::PassManager;
     }
   }
 
-  OrderedNode* CachedNZCV = {};
-  uint32_t PossiblySetNZCVBits = 0;
+  OrderedNode* CachedWriteNZCV{};
+  OrderedNode* CachedReadNZCV{};
```

Comment:
I don't know if this is the right approach. The existing code implements a standard pattern *local value numbering* ... would it be practical to simply augment with a bool `NZCVDirty` instead of duplicating the node?

Reply:
I can switch over to a dirty flag instead

Reply:
Thanks, I suspect that will make more sense, though I'm not super confident tbh

Reply:
Rewrote to use a dirty flag to track the writes instead.

Reply:
Thanks, the new commit makes a lot more sense to me! ... CI doesn't seem to agree yet but once it does this will get my approval :+1: